x/subscription/types/message_buy.go

```diff
@@ -58,6 +59,9 @@ func (msg *MsgBuy) ValidateBasic() error {
 	if msg.Duration == 0 || msg.Duration > MAX_SUBSCRIPTION_DURATION {
 		return sdkerrors.Wrapf(ErrInvalidParameter, "invalid subscription duration (%s)", msg.Index)
 	}
+	if msg.AutoRenewal && msg.AdvancePurchase {
```

Comment:
why not?

Reply:
Because it's 2 different actions, that call different code on `subscription.go`.

Mixing them together does not really make any sense, and I want to let know the user about that

Reply:
why?
what if i want to advance purchase and after that to make sure my subscription autrenewed?
how does it contradict?

Reply:
Auto renewed to what? To the future subscription, or the current one? It's ambiguous...